node/internal.ts

```diff
@@ -282,12 +282,12 @@ export function _command(command: string, properties: any, message: string) {
     _writeLine(taskCmd.toString());
 }
 
-export function _warning(message: string): void {
-    _command('task.issue', { 'type': 'warning' }, message);
+export function _warning(message: string, source?: string): void {
+    _command('task.issue', { 'type': 'warning', 'source': source }, message);
 }
 
-export function _error(message: string): void {
-    _command('task.issue', { 'type': 'error' }, message);
+export function _error(message: string, source?: string): void {
```

Comment:
What sources are we expected to see?
Probably it's better to use enum/template type

Reply:
Expected sources are something like 'customerscript', 'taskinternal'

Reply:
Okay, probably you're right. I'll re-implement this thing, no big deal